src/contexts/NetworkFeeProvider.tsx

```diff
@@ -13,22 +15,87 @@ import { useConnectionContext } from '@src/contexts/ConnectionProvider';
+import { FundTxHandler } from '@src/background/services/gasless/handlers/fundTx';
+import { GetGaslessEligibilityHandler } from '@src/background/services/gasless/handlers/getGaslessEligibility';
+import { FetchAndSolveChallengeHandler } from '@src/background/services/gasless/handlers/fetchAndSolveChallange';
+import { filter, map } from 'rxjs';
+import { gaslessChallangeUpdateEventListener } from '@src/background/services/gasless/events/gaslessChallangeUpdateListener';
+import { AddressLike, TransactionRequest } from 'ethers';
+import { useFeatureFlagContext } from './FeatureFlagsProvider';
+import { FeatureGates } from '@src/background/services/featureFlags/models';
+import { SetDefaultStateValuesHandler } from '@src/background/services/gasless/handlers/setDefaultStateValues';
+import { GaslessPhase } from '@src/background/services/gasless/model';
 
 const NetworkFeeContext = createContext<{
   networkFee: NetworkFee | null;
   getNetworkFee: (caipId: string) => Promise<NetworkFee | null>;
+  fetchAndSolveGaslessChallange: () => Promise<any | null>;
+  gaslessFundTx: ({
+    data,
+    fromAddress,
+  }: {
+    data: TransactionRequest;
+    fromAddress: string;
+  }) => Promise<string | undefined>;
+  isGaslessOn: boolean;
+  setIsGaslessOn: Dispatch<SetStateAction<boolean>>;
+  isGaslessEligible: boolean | null;
+  fundTxHex: string;
+  setGaslessDefaultValues: () => Promise<any | null>;
+  gaslessPhase: GaslessPhase | null;
+  setGaslessPhase: Dispatch<SetStateAction<GaslessPhase | null>>;
+  setGaslessEligibility: (
+    chainId: string | number,
+    fromAddress?: AddressLike | null,
+    nonce?: number | null,
+  ) => Promise<void>;
 }>({
   networkFee: null,
   async getNetworkFee() {
     return null;
   },
+  async fetchAndSolveGaslessChallange() {
+    return null;
+  },
+
+  async gaslessFundTx() {
+    return undefined;
+  },
+  isGaslessOn: false,
+  setIsGaslessOn() {
+    return null;
+  },
+  isGaslessEligible: null,
+
+  fundTxHex: '',
+
+  async setGaslessDefaultValues() {
+    return null;
+  },
+
+  gaslessPhase: null,
+  setGaslessPhase() {
+    return null;
+  },
+  async setGaslessEligibility() {
+    return;
+  },
 });
 
 export function NetworkFeeContextProvider({ children }: { children: any }) {
-  const { request } = useConnectionContext();
+  const { request, events } = useConnectionContext();
   const { network } = useNetworkContext();
   const [fee, setFee] = useState<NetworkFee | null>(null);
   const [iteration, setIteration] = useState(0);
+  const [challengeHex, setChallengeHex] = useState('');
+  const [solutionHex, setSolutionHex] = useState('');
+  const [isGaslessOn, setIsGaslessOn] = useState(false);
+  const [isGaslessEligible, setIsGaslessEligible] = useState<boolean | null>(
+    null,
+  );
+  const [fundTxHex, setFundTxHex] = useState('');
+  const { featureFlags } = useFeatureFlagContext();
+  const [gaslessPhase, setGaslessPhase] = useState<GaslessPhase | null>(null);
```

Comment:
Instead of allowing null, it would be nicer to have a `GaslessPhase.READY` value. This would mean no null checks and the state is more explicit.